### PR TITLE
github: refactor verify workflow

### PR DESCRIPTION
**Pull request by marquiz**
Change the verify workflow to run on PRs only. Add concurrency rule so that in-progress runs for a PR are canceled if the PR is updated before their completion.

Add a nightly verify workflow that runs on the master branch every day at 2:30 UTC.

### CODE REVIEW

File: .github/workflows/common-verify.yaml [hunk 1]
```diff
@@ -1,13 +1,18 @@
...
 
-  build:
-    name: Build
+on:
+  workflow_call:
+    inputs:
+      publish:
+        default: false
+        required: false
+        type: boolean
```
Comment (klihub) [inline .github/workflows/common-verify.yaml:9]: Why are these needed ?

Comment (marquiz) [reply .github/workflows/common-verify.yaml]: No use, stale leftovers from development. Now removed

**Comment by marquiz** [comment]
@klihub this needs #1038

**Review by klihub — APPROVED** [review]

**Comment by marquiz** [comment]
Rebased

**Comment by marquiz** [comment]
ping @klihub

**Review by klihub — APPROVED** [review]
